packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,287 @@
+import { openDatabase } from 'expo-sqlite';
+import { ParameterizedStatement } from '../src/common/types';
+import { HelperStack } from 'typedoc/dist/lib/output/utils/resources/helpers';
+
+// mock out the underlying implementation and only keep openDatabase().
+jest.mock('expo-sqlite', () => {
+	return {
+		__esModule: true,
+		openDatabase: jest.fn(() => mockDatabase),
+	};
+});
+
+// mock out the underlying implementation
+jest.mock('expo-file-system', () => 'File System');
+
+const mockDatabase = {
+	readTransaction: jest.fn(),
+	transaction: jest.fn(),
+};
+
+const mockTransaction = {
+	executeSql: jest.fn(),
+};
+
+const mockResultSet = jest.fn();
+
+const MOCK_SQL_ERROR = {
+	code: 400,
+	message: 'Error performing the transaction',
+};
+
+describe('ExpoSQLiteAdapter', () => {
+	const ExpoSQLiteAdapterInstance = new ExpoSQLiteDatabase();
+	ExpoSQLiteAdapterInstance.init();
+
+	beforeEach(() => {
+		mockDatabase.readTransaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
+		mockDatabase.transaction.mockImplementation(func => {
+			func(mockTransaction);
+		});
+
+		// mockImplementation of executeSql first two calls should call successCallback, followed by errorCallback
+		// and remaining all execution wil call successCallback
+		// intended to check behavior if promises are resolved and rejected correctly
+		mockTransaction.executeSql
+			.mockImplementationOnce(
+				(statement, params, successCallback, errorCallback) => {
+					successCallback(mockTransaction, mockResultSet());
+				}
+			)
+			.mockImplementationOnce(
+				(statement, params, successCallback, errorCallback) => {
+					successCallback(mockTransaction, mockResultSet());
+				}
+			)
+			.mockImplementationOnce(
+				(statement, params, successCallback, errorCallback) => {
+					errorCallback(mockTransaction, MOCK_SQL_ERROR);
+				}
+			)
+			// set default mock function after three calls
+			.mockImplementation(
+				(statement, params, successCallback, errorCallback) => {
+					successCallback(mockTransaction, mockResultSet());
+				}
+			);
+
+		mockResultSet
+			.mockImplementationOnce(() => {
+				return {
+					rows: {
+						_array: MOCK_RESULT_ARRAY_1,
+					},
+				};
+			})
+			.mockImplementationOnce(() => {
+				return {
+					rows: {
+						_array: [],
+					},
+				};
+			})
+			// set default mock function after two calls
+			.mockImplementation(() => {
+				return {
+					rows: {
+						_array: MOCK_RESULT_ARRAY_2,
+					},
+				};
+			});
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('getAll() Tests', () => {
+		beforeAll(() => {
+			jest.resetAllMocks();
+		});
+		it('should resolve promise and return whole result array', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.getAll(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).then(result => {
+				expect(result).toEqual(MOCK_RESULT_ARRAY_1);
+			});
+		});
```

Comment:
You can simplify this for readers if you use `async/await` with `.resolves/.rejects`
See: https://jestjs.io/docs/asynchronous#asyncawait

```suggestion
		it('should resolve promise and return whole result array', async () => {
			await expect(
				ExpoSQLiteAdapterInstance.getAll(DEFAULT_STATEMENT, DEFAULT_STATEMENT_PARAMS)
			).resolves.toStrictEqual(MOCK_RESULT_ARRAY_1);
		});
```

Ditto for other functions/tests as well

Reply:
Thanks. I'll simplify it.